python/ray/data/dataset_pipeline.py

```diff
@@ -1,1418 +0,0 @@
-import itertools
-import logging
-import sys
-import time
-import warnings
-from typing import (
-    TYPE_CHECKING,
-    Any,
-    Callable,
-    Dict,
-    Iterable,
-    Iterator,
-    List,
-    Optional,
-    Union,
-)
-
-import numpy as np
-
-import ray
-from ray.air.util.data_batch_conversion import BlockFormat
-from ray.data._internal.block_batching import batch_block_refs
-from ray.data._internal.block_list import BlockList
-from ray.data._internal.compute import ComputeStrategy
-from ray.data._internal.iterator.pipelined_iterator import PipelinedDataIterator
-from ray.data._internal.pipeline_executor import (
-    PipelineExecutor,
-    PipelineSplitExecutorCoordinator,
-)
-from ray.data._internal.plan import ExecutionPlan
-from ray.data._internal.stats import DatasetPipelineStats, DatasetStats
-from ray.data.block import (
-    Block,
-    DataBatch,
-    UserDefinedFunction,
-    _apply_strict_mode_batch_format,
-)
-from ray.data.context import DataContext
-from ray.data.dataset import Dataset
-from ray.data.datasource import Datasource
-from ray.data.datasource.file_based_datasource import (
-    BlockWritePathProvider,
-    DefaultBlockWritePathProvider,
-)
-from ray.data.iterator import DataIterator
-from ray.types import ObjectRef
-from ray.util.annotations import Deprecated, DeveloperAPI
-from ray.util.scheduling_strategies import NodeAffinitySchedulingStrategy
-
-if sys.version_info >= (3, 8):
-    from typing import Literal
-else:
-    from typing_extensions import Literal
-
-if TYPE_CHECKING:
-    import pandas
-    import pyarrow
-    import tensorflow as tf
-    import torch
-
-    from ray.data._internal.torch_iterable_dataset import TorchTensorBatchType
-
-
-logger = logging.getLogger(__name__)
-
-
-@Deprecated
-class DatasetPipeline:
-    """Implements a pipeline of Datasets.
-
-    DatasetPipelines implement pipelined execution. This allows for the
-    overlapped execution of data input (e.g., reading files), computation
-    (e.g. feature preprocessing), and output (e.g., distributed ML training).
-
-    A DatasetPipeline can be created by either repeating a Dataset
-    (``ds.repeat(times=None)``), by turning a single Dataset into a pipeline
-    (``ds.window(blocks_per_window=10)``), or defined explicitly using
-    ``DatasetPipeline.from_iterable()``.
```

Comment:
Given people may still use DatasetPipeline via `DatasetPipeline.from_iterable()`, let's leave a deprecation warning in `DatasetPipeline.from_iterable()`, and delete everything else.